enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java

```diff
@@ -334,11 +335,23 @@ private boolean catchUpWithMaster( UpdatePuller updatePuller ) throws IllegalArg
 
     private URI createHaURI( URI me, Server<?,?> server )
     {
-        String hostString = ServerUtil.getHostString( server.getSocketAddress() );
-        int port = server.getSocketAddress().getPort();
+        InetSocketAddress serverSocketAddress = server.getSocketAddress();
+        String hostString = ServerUtil.getHostString( serverSocketAddress );
+
+        String host = isWildcard( hostString ) ? me.getHost() : hostString;
+
+        if ( host.contains( ":" ) && !host.contains( "[" ) )
+        {
+            host = "[" + host + "]";
+        }
+
         InstanceId serverId = config.get( ClusterSettings.server_id );
-        String host = hostString.contains( HighAvailabilityModeSwitcher.INADDR_ANY ) ? me.getHost() : hostString;
-        return URI.create( "ha://" + host + ":" + port + "?serverId=" + serverId );
+        return URI.create( "ha://" + host + ":" + serverSocketAddress.getPort() + "?serverId=" + serverId );
+    }
+
+    private static boolean isWildcard( String hostString )
```

Comment:
It is actually subtly different. One checks a String straight out of config where I expect the [ ] format whereas the other one checks an address without the square brackets. I know it is crazy to have this code in this place this far down in the stack and duplicated, but I do not want to attempt a refactoring and cleanup.